In [1]:
import pandas as pd
import numpy as np
import matplotlib as plt

#To read the dataset for analysis

In [2]:
data = pd.read_csv("crop_production.csv")

In [3]:
data.head()  #to print the first five rows

State_Name District_Name  Crop_Year       Season  \
0  Andaman and Nicobar Islands      NICOBARS       2000  Kharif        
1  Andaman and Nicobar Islands      NICOBARS       2000  Kharif        
2  Andaman and Nicobar Islands      NICOBARS       2000  Kharif        
3  Andaman and Nicobar Islands      NICOBARS       2000  Whole Year    
4  Andaman and Nicobar Islands      NICOBARS       2000  Whole Year    

                  Crop    Area  Production  
0             Arecanut  1254.0      2000.0  
1  Other Kharif pulses     2.0         1.0  
2                 Rice   102.0       321.0  
3               Banana   176.0       641.0  
4            Cashewnut   720.0       165.0

In [4]:
data.tail() #print last five rows

State_Name District_Name  Crop_Year       Season       Crop  \
246086  West Bengal       PURULIA       2014  Summer            Rice   
246087  West Bengal       PURULIA       2014  Summer         Sesamum   
246088  West Bengal       PURULIA       2014  Whole Year   Sugarcane   
246089  West Bengal       PURULIA       2014  Winter            Rice   
246090  West Bengal       PURULIA       2014  Winter         Sesamum   

            Area  Production  
246086     306.0       801.0  
246087     627.0       463.0  
246088     324.0     16250.0  
246089  279151.0    597899.0  
246090     175.0        88.0

In [13]:
data.shape
#246091 rows, 7 columns

(246091, 7)

In [5]:
data.describe()

Crop_Year          Area    Production
count  246091.000000  2.460910e+05  2.423610e+05
mean     2005.643018  1.200282e+04  5.825034e+05
std         4.952164  5.052340e+04  1.706581e+07
min      1997.000000  4.000000e-02  0.000000e+00
25%      2002.000000  8.000000e+01  8.800000e+01
50%      2006.000000  5.820000e+02  7.290000e+02
75%      2010.000000  4.392000e+03  7.023000e+03
max      2015.000000  8.580100e+06  1.250800e+09

In [6]:
data.isnull().sum()

State_Name          0
District_Name       0
Crop_Year           0
Season              0
Crop                0
Area                0
Production       3730
dtype: int64

#Only the production column has missing values numbering 3730

In [7]:
data["Area"]

0           1254.0
1              2.0
2            102.0
3            176.0
4            720.0
            ...   
246086       306.0
246087       627.0
246088       324.0
246089    279151.0
246090       175.0
Name: Area, Length: 246091, dtype: float64

In [8]:
data["Area"].mean()

12002.820863704896

#Only production column has 3730 nulls

In [9]:
pd.Series(data["District_Name"].unique())

0                      NICOBARS
1      NORTH AND MIDDLE ANDAMAN
2                SOUTH ANDAMANS
3                     ANANTAPUR
4                      CHITTOOR
                 ...           
641              MEDINIPUR EAST
642              MEDINIPUR WEST
643                 MURSHIDABAD
644                       NADIA
645                     PURULIA
Length: 646, dtype: object

In [10]:
len(data["District_Name"].unique())
#646 district names 

646

In [11]:
sea = data["Season"].unique()
pd.Series(sea)

0    Kharif     
1    Whole Year 
2    Autumn     
3    Rabi       
4    Summer     
5    Winter     
dtype: object

In [12]:
len(data["Season"].unique())
#only 6 seasons are there in

6

#the  below shows that the dataset has 246091 rows and 7 columns.

In [14]:
#crop years in ascending order
crop_year = data["Crop_Year"].unique()
crop_year.sort()
crop_year

array([1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,
       2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015], dtype=int64)

#The dataset's info was gotten from year 1997 to 2015

In [15]:
state = data["State_Name"].unique()
print(f"{state}, \n\nThere are {len(state)} states considered in the dataset.")

['Andaman and Nicobar Islands' 'Andhra Pradesh' 'Arunachal Pradesh'
 'Assam' 'Bihar' 'Chandigarh' 'Chhattisgarh' 'Dadra and Nagar Haveli'
 'Goa' 'Gujarat' 'Haryana' 'Himachal Pradesh' 'Jammu and Kashmir '
 'Jharkhand' 'Karnataka' 'Kerala' 'Madhya Pradesh' 'Maharashtra' 'Manipur'
 'Meghalaya' 'Mizoram' 'Nagaland' 'Odisha' 'Puducherry' 'Punjab'
 'Rajasthan' 'Sikkim' 'Tamil Nadu' 'Telangana ' 'Tripura' 'Uttar Pradesh'
 'Uttarakhand' 'West Bengal'], 

There are 33 states considered in the dataset.


In [16]:
crop_num = data["Crop"].unique()
print(f"{len(crop_num)} crops")

124 crops


In [17]:
#data showing only state West Bangal and its correspondence
data.loc[data["State_Name"] == "West Bengal"]

State_Name       District_Name  Crop_Year       Season       Crop  \
236478  West Bengal  24 PARAGANAS NORTH       1997  Autumn            Rice   
236479  West Bengal  24 PARAGANAS NORTH       1997  Kharif            Jute   
236480  West Bengal  24 PARAGANAS NORTH       1997  Kharif           Mesta   
236481  West Bengal  24 PARAGANAS NORTH       1997  Kharif            Urad   
236482  West Bengal  24 PARAGANAS NORTH       1997  Rabi              Gram   
...             ...                 ...        ...          ...        ...   
246086  West Bengal             PURULIA       2014  Summer            Rice   
246087  West Bengal             PURULIA       2014  Summer         Sesamum   
246088  West Bengal             PURULIA       2014  Whole Year   Sugarcane   
246089  West Bengal             PURULIA       2014  Winter            Rice   
246090  West Bengal             PURULIA       2014  Winter         Sesamum   

            Area  Production  
236478   15017.0     33840.0  
236479   52568.0    677510.0  
236480     690.0      6710.0  
236481    1992.0      1310.0  
236482     267.0       160.0  
...          ...         ...  
246086     306.0       801.0  
246087     627.0       463.0  
246088     324.0     16250.0  
246089  279151.0    597899.0  
246090     175.0        88.0  

[9613 rows x 7 columns]

In [19]:
data["Crop"].max()

'other oilseeds'

In [22]:
#the crops planted 
crops = data["Crop"].unique()
d.sort()
d

array(['Apple', 'Arcanut (Processed)', 'Arecanut', 'Arhar/Tur',
       'Ash Gourd', 'Atcanut (Raw)', 'Bajra', 'Banana', 'Barley', 'Bean',
       'Beans & Mutter(Vegetable)', 'Beet Root', 'Ber', 'Bhindi',
       'Bitter Gourd', 'Black pepper', 'Blackgram', 'Bottle Gourd',
       'Brinjal', 'Cabbage', 'Cardamom', 'Carrot', 'Cashewnut',
       'Cashewnut Processed', 'Cashewnut Raw', 'Castor seed',
       'Cauliflower', 'Citrus Fruit', 'Coconut ', 'Coffee', 'Colocosia',
       'Cond-spcs other', 'Coriander', 'Cotton(lint)', 'Cowpea(Lobia)',
       'Cucumber', 'Drum Stick', 'Dry chillies', 'Dry ginger', 'Garlic',
       'Ginger', 'Gram', 'Grapes', 'Groundnut', 'Guar seed', 'Horse-gram',
       'Jack Fruit', 'Jobster', 'Jowar', 'Jute', 'Jute & mesta', 'Kapas',
       'Khesari', 'Korra', 'Lab-Lab', 'Lemon', 'Lentil', 'Linseed',
       'Litchi', 'Maize', 'Mango', 'Masoor', 'Mesta', 'Moong(Green Gram)',
       'Moth', 'Niger seed', 'Oilseeds total', 'Onion', 'Orange',
       'Other  Rabi pulses

In [23]:
data.loc[data["Crop"] == "other oilseeds"].count()

State_Name       633
District_Name    633
Crop_Year        633
Season           633
Crop             633
Area             633
Production       594
dtype: int64

In [27]:
#to know the frequency of each crops
data["Crop"].value_counts() 

#Rice has the highest crop planted....and "other dry fruits" is lowest

Rice                 15104
Maize                13947
Moong(Green Gram)    10318
Urad                  9850
Sesamum               9046
                     ...  
Plums                    6
Coffee                   6
Peach                    4
Apple                    4
Other Dry Fruit          1
Name: Crop, Length: 124, dtype: int64